flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -701,12 +701,14 @@ public boolean contains(ConfigOption<?> configOption) {
 			if (this.confData.containsKey(configOption.key())) {
 				return true;
 			}
-			else if (configOption.hasDeprecatedKeys()) {
-				// try the deprecated keys
-				for (String deprecatedKey : configOption.deprecatedKeys()) {
-					if (this.confData.containsKey(deprecatedKey)) {
-						LOG.warn("Config uses deprecated configuration key '{}' instead of proper key '{}'",
-							deprecatedKey, configOption.key());
+			else if (configOption.hasFallbackKeys()) {
+				// try the fallback keys
+				for (FallbackKey fallbackKey : configOption.fallbackKeys()) {
+					if (this.confData.containsKey(fallbackKey.getKey())) {
+						if (fallbackKey.isDeprecated()) {
```

Comment:
Maybe it would be a good idea to log on `info` if we fall back to another key if `fallbackKey.isDeprecated == false`. That way, the user knows from which option it has read the value.

Reply:
agree